.grit/patterns/join_nested_withs.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Join nested with statements
+---
+
+Join nested with statements into a single one. Rule [SIM117](https://github.com/MartinThoma/flake8-simplify/issues/35) from [flake8-simplify](https://github.com/MartinThoma/flake8-simplify).
+
+Limitations:
+* Only two nested with statements are joined.
+
+```grit
+engine marzano(0.1)
+language python
+
+
+`
+with $context1 as $as1:
+    with $context2 as $as2:
+        $with_body
+` => `with $context1 as $as1, $context2 as $as2:
+    $with_body`
+```
+
+# Join with statements
+
+```python
+with open("file1.txt") as f1:
+    with open("file2.txt", "r+") as f2:
+        pass
+
+# TODO: should be joined into a single with
```

Comment:
Shouldn't we also handle this case?

```
with A() as a, B() as b:
  with C() as C:
    pass
```

Once we do, this case is easily handled.

Reply:
With the latest changes,

```
with A() as a, B() as b:
    with C() as c:
        pass
```

Is correctly converted to:

```
with A() as a, B() as b, C() as c:
    pass
```

But I still can't join three `with`s one inside the other:

```
with A() as a:
    with B() as b:
        with C() as c:
            pass
```

Reply:
Thanks, that's somewhat to be expected without recursive pattern support.

Reply:
Ok!